pkg/aws/eni/routing/routing.go

```diff
@@ -122,6 +129,73 @@ func Install(ip net.IP, info *RoutingInfo, mtu int, masq bool) error {
 	return nil
 }
 
+// Delete removes the ingress and egress rules that control traffic for
+// endpoints. Note that the routes within these rules are not deleted as they
+// can be reused when another endpoint is created on the same node. The reason
+// for this is that each ENI device has an index called ifindex which is used
+// as the table ID when these rules are created. The routes are created inside
+// a table with this ID, and because this table ID is tied to the ENI device,
+// it's stable to rely on and therefore can be reused.
+func Delete(ip net.IP) error {
+	ipWithMask := net.IPNet{
+		IP:   ip,
+		Mask: net.CIDRMask(32, 32),
```

Comment:
I got a bit caught up on this, as it's clearly assuming IPv4 but there's no validation within the function.

I guess there's a couple of potential mitigations. Simplest would be adding an extra line at the bottom of the comment that states that this is only used for IPv4.

Alternatively we could use logic similar to this:

https://github.com/cilium/cilium/blob/97f8299d6c1a0da4a3f23c296f1e719c72e94c35/pkg/ipcache/ipcache.go#L137-L144

But then, the caller is explicictly only passing an IPv4 address down so it's not so obviously necessary to handle this. Probably fine as-is?

Reply:
ENI mode is only implemented with IPv4 support, which is also listed in the docs: https://docs.cilium.io/en/v1.7/gettingstarted/aws-eni/?highlight=eni%20ipv4#limitations

We can add validation to the function to only expect IPv4 for now, but that doesn't seem like the right level of validation. I assumed that we would validate ENI + IPv4, but I don't see that enforced anywhere in the code. That might be the right thing to do instead. What do you think?

Reply:
That sounds reasonable.